core/domain/classifier_services.py

```diff
@@ -347,6 +348,75 @@ def mark_training_job_complete(job_id):
                                            feconf.TRAINING_JOB_STATUS_COMPLETE)
 
 
+def mark_training_job_failed(job_id):
+    """Updates the training job's status to failed.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_FAILED)
+
+
+def mark_training_job_pending(job_id):
+    """Updates the training job's status to pending.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_PENDING)
+
+
+def update_failed_jobs(training_jobs):
+    """Updates status of job_models to failed.
+
+    Args:
+        training_jobs: list. List of jobs to be updated as failed.
+    """
+    for training_job in training_jobs:
+        mark_training_job_failed(training_job.job_id)
+
+
+def fetch_next_job():
+    """Gets next job model in the job queue.
+
+    Returns:
+        ClassifierTrainingJobModel. Domain object of the next training Job.
+    """
+    classifier_training_jobs = []
+    cursor = None
+    valid_jobs = []
+    failed_jobs = []
+
+    while len(valid_jobs) == 0:
```

Comment:
(1) As of now emailing admin sounds good (unless we have a lot of failing jobs). We can also go with monitoring but I am not sure how to do so. May be store the failed jobs IDs somewhere in some model? Then admin can simply check this model from dashboard to find failed jobs?
Well, if you think that emails are better than dashboard then I'd be more than happy to work on that (in a separate issue / PR). So let me know which one do you think is more helpful? I think emails are more helpful and direct. 

(2) Yep we will have to check pending jobs again and again before TTL. OK, so now I understand what were you trying to say. So when the job is fetched increment last_checked_time by TTL and then when we query jobs only query for those jobs which have last_checked_time < utcnow(), right? This way we can avoid fetching valid pending jobs. That's great suggestion.